src/DicomWebClient/API/IDicomWebClient.cs

```diff
@@ -28,5 +29,36 @@ public interface IDicomWebClient
         /// Provides DICOMweb WADO services for retrieving studies, series, instances, frames and bulkdata.
         /// </summary>
         IWadoService Wado { get; }
+
+        /// <summary>
+        /// Provides DICOMweb QIDO services for querying a remote server for studies.
+        /// </summary>
+        IQidoService Qido { get; }
+
+        /// <summary>
+        /// Provides DICOMweb STOW services for storing DICOM instances.
+        /// </summary>
+        IStowService Stow { get; }
+
+        /// <summary>
+        /// Configures the service URI of the DICOMweb service.
+        /// </summary>
+        /// <param name="uriRoot">Base URL of the DICOMweb server.</param>
+        /// <param name="wadoUrlPrefix">Optional URL path prefix for WADO RESTful services.</param>
+        /// <param name="qidoUrlPrefix">Optional URL path prefix for QIDO RESTful services.</param>
+        /// <param name="stowUrlPrefix">Optional URL path prefix for STOW RESTful services.</param>
+        /// <param name="deleteUrlPrefix">Optional URL path prefix for DELETE RESTful services.></param>
+        void ConfigureServiceUris(
```

Comment:
I see an issue with this API method as it assumes that all the DICOMWeb service endpoints are under the same base URL. May be true in some cases, however, there are already cases where the studies are retrieved from on DICOM archive whereas the generated DICOM instances are sent to another achieve. So the "gets", WADO and QIDO, may be from the same Web server, but the "puts" are likely to a different server.
Also, this API does not allow specifying service specific attributes, e.g. credential, timeout, etc.
So, it is more user friendly to have a DICOMWeb service base class, and this API method will just pass in the concrete instances for WASO, QIDO, and STOW RS, null if not applicable.

Reply:
In cases where the URL is different, the user will instantiate a separate instance of DICOMWebClient. 
For authentication, timeout, and other client attributes, these can be pre-configured in the instance of `HttpClient` that is passed into the `DicomWebClient` constructor.